src/picongpu/include/particles/pusher/particlePusherReducedLandauLifshitz.hpp

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright 2013-2015 Axel Huebl, Heiko Burau, Rene Widera, Richard Pausch
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#pragma once
+
+#include "types.h"
+#include "simulation_defines.hpp"
+#include "math/Vector.hpp"
+#include "math/RungeKutta.hpp"
+
+
+namespace picongpu
+{
+namespace particlePusherReducedLandauLifshitz
+{
+
+template<class Velocity, class Gamma>
+struct Push
+{
+  /* this is an optional extention for sub-sampling pushes that enables grid to particle interpolation
+   * for particle positions outside the super cell in one push
+   */
+  typedef typename PMacc::math::CT::make_Int<simDim,1>::type LowerMargin;
+  typedef typename PMacc::math::CT::make_Int<simDim,1>::type UpperMargin;
+
+  template<typename T_FunctorFieldB, typename T_FunctorFieldE, typename T_Pos, typename T_Mom,
+           typename T_Mass, typename T_Charge, typename T_Weighting >
+  __host__ DINLINE void operator()(
+                                   const T_FunctorFieldB functorBField, /* at t=0 */
+                                   const T_FunctorFieldE functorEField, /* at t=0 */
+                                   T_Pos& pos, /* at t=0 */
+                                   T_Mom& mom, /* at t=-1/2 */
+                                   const T_Mass mass,
+                                   const T_Charge charge,
+                                   const T_Weighting weighting)
+  {
+    typedef T_FunctorFieldB TypeBFieldFunctor;
+    typedef T_FunctorFieldE TypeEFieldFunctor;
+    typedef T_Pos TypePosition;
+    typedef T_Mom TypeMomentum;
+    typedef T_Mass TypeMass;
+    typedef T_Charge TypeCharge;
+    typedef T_Weighting TypeWeighting;
+
+    const float_X deltaT = DELTA_T;
+    typedef PMacc::math::Vector<picongpu::float_X, 6> VariableType;
+    VariableType var;
+    
+    // transfer position
+    for(uint i=0; i<picongpu::simDim; ++i)
+      var[i] = pos[i];
```

Comment:
even if RungeKutta4 should not use it, it is very bad practice to fiddle around with unitialized vars.

either use less components (which is way more work) or initialize unused components savely with zero.